tests/dialects/test_tsql.py

```diff
@@ -631,3 +637,29 @@ def test_current_user(self):
             "SUSER_SNAME()",
             write={"spark": "CURRENT_USER()"},
         )
+        self.validate_all(
+            "SYSTEM_USER()",
+            write={"spark": "CURRENT_USER()"},
+        )
+        self.validate_all(
+            "SYSTEM_USER",
+            write={"spark": "CURRENT_USER()"},
+        )
+
+    def test_join_hint(self):
```

Comment:
We should also test the generation of hints for T-SQL (i.e. they should be preserved).

Reply:
I did not see the value in generating the hints, due to the fact you are usually transpiling towards a different dialect and those hints arent usable in those cases. Thats why I just fixed the parsing part. Is that ok with you?

Reply:
We should generate them because someone might want to use SQLGlot to format T-SQL, and so we can't break their queries by discarding those hints.

Reply:
Ah ok, will take a look at that one.

Reply:
Thanks, I think it should just involve handling the "hints" arg in `join_sql`.

Reply:
Still, not in the generic class as they are only supported in T-SQL. You would only want to output when the read dialect is the same as the write dialect. Or do you have a good idea on this?

Reply:
One idea is to create a new class variable called something like `WITH_JOIN_HINT` that will control whether or not the `hint` arg will be used, and set its default value to be `False` (T-SQL will set that to be `True` -- see other recently added variables for reference).